MueLu: build error w/ Zoltan but w/o Zoltan2

## Bug Report
@trilinos/muelu 

### Description

With `Zoltan` enabled, but `Zoltan2` disabled, `packages/muelu/src/Interface/MueLu_ParameterListInterpreter_def.hpp` produces a build error due to some `#ifdef` logic related to Zoltan/Zoltan2 activation:

PR #7263 addresses this. This issue is rather intended to note, that the PR autotester did not catch this.

### Steps to Reproduce
1. SHA1: de15ca5352bdd31e94a2ba6af1a1b56cefe546da
1. Configure script:
    <details><summary>CMake command></summary>

    ```
    $CMAKE_COMMAND \
    -GNinja -D Trilinos_ENABLE_Fortran:BOOL=OFF \
    -D CMAKE_BUILD_TYPE:STRING="RELEASE" \
    -D CMAKE_CXX_COMPILER:FILEPATH="$MPI_BIN_DIR/mpic++" \
    -D CMAKE_C_COMPILER:FILEPATH="$MPI_BIN_DIR/mpicc" \
    -D CMAKE_Fortran_COMPILER:FILEPATH="$MPI_BIN_DIR/mpif90" \
    -D CMAKE_CXX_FLAGS:STRING="-fdiagnostics-color=always" \
    -D CMAKE_INSTALL_PREFIX:STRING=`pwd` \
    -D CMAKE_VERBOSE_MAKEFILE:BOOL=OFF \
    -D CMAKE_COLOR_MAKEFILE:BOOL=ON \
    -D BUILD_SHARED_LIBS:BOOL=OFF \
    \
    -D Trilinos_ASSERT_MISSING_PACKAGES=OFF \
    -D Trilinos_VERBOSE_CONFIGURE:BOOL=OFF \
    -D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF \
    -D Trilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
    -D Trilinos_ENABLE_ALL_PACKAGES:BOOL=OFF \
    -D Trilinos_ENABLE_TESTS:BOOL=OFF \
    -D Trilinos_ENABLE_EXAMPLES:BOOL=OFF \
    \
    -D Trilinos_ENABLE_Amesos:BOOL=ON \
    -D Trilinos_ENABLE_Anasazi:BOOL=ON \
    -D Trilinos_ENABLE_AztecOO:BOOL=ON \
    -D Trilinos_ENABLE_Belos:BOOL=ON \
    -D Trilinos_ENABLE_Epetra:BOOL=ON \
      -D Epetra_HIDE_DEPRECATED_CODE:BOOL=ON \
    -D Trilinos_ENABLE_EpetraExt:BOOL=ON \
      -D EpetraExt_BUILD_GRAPH_REORDERINGS:BOOL=ON \
      -D EpetraExt_ENABLE_HDF5:BOOL=OFF \
    -D Trilinos_ENABLE_Intrepid:BOOL=ON \
    -D Trilinos_ENABLE_Ifpack:BOOL=ON \
    -D Trilinos_ENABLE_Isorropia:BOOL=ON \
    -D Trilinos_ENABLE_Kokkos:BOOL=ON \
      -D Kokkos_ENABLE_CUDA:BOOL=OFF \
      -D Kokkos_ENABLE_DEPRECATED_CODE:BOOL=OFF \
    -D Trilinos_ENABLE_ML:BOOL=ON \
    -D Trilinos_ENABLE_MueLu:BOOL=ON \
      -D MueLu_ENABLE_DEPRECATED_CODE:BOOL=OFF \
      -D MueLu_ENAABLE_Kokkos_Refactor:BOOL=OFF \
    -D Trilinos_ENABLE_NOX:BOOL=ON \
    -D Trilinos_ENABLE_Sacado:BOOL=ON \
    -D Trilinos_ENABLE_SEACASExodus:BOOL=ON \
    -D Trilinos_ENABLE_SEACASNemesis:BOOL=OFF \
    -D Trilinos_ENABLE_Shards:BOOL=ON \
    -D Trilinos_ENABLE_Teuchos:BOOL=ON \
      -D Teuchos_HIDE_DEPRECATED_CODE:BOOL=ON \
    -D Trilinos_ENABLE_Teko:BOOL=ON \
    -D Trilinos_ENABLE_Tpetra:BOOL=ON \
      -D Tpetra_ENABLE_DEPRECATED_CODE:BOOL=OFF \
      -D Tpetra_INST_INT_INT:BOOL=ON \
    -D Trilinos_ENABLE_Xpetra:BOOL=ON \
      -D Xpetra_ENABLE_DEPRECATED_CODE:BOOL=OFF \
    -D Trilinos_ENABLE_Zoltan:BOOL=ON \
    -D Trilinos_ENABLE_Zoltan2:BOOL=OFF \
    \
    -D TPL_ENABLE_Boost:BOOL=ON \
    -D TPL_ENABLE_MPI:BOOL=ON \
    -D TPL_ENABLE_ParMETIS:BOOL=ON \
      -D ParMETIS_INCLUDE_DIRS:PATH="$TPLDIR/include" \
      -D ParMETIS_LIBRARY_DIRS:PATH="$TPLDIR/lib" \
    -D TPL_ENABLE_Pthread:BOOL=ON \
    -D TPL_ENABLE_UMFPACK:BOOL=ON \
      -D UMFPACK_INCLUDE_DIRS:FILEPATH="$TPLDIR/include" \
      -D UMFPACK_LIBRARY_DIRS:FILEPATH="$TPLDIR/lib" \
    -D TPL_ENABLE_Netcdf:BOOL=ON \
    -D TPL_ENABLE_MATLAB:BOOL=OFF \
    -D TPL_ENABLE_SuperLU:BOOL=OFF \
    -D TPL_ENABLE_SuperLUDist:BOOL=ON \
      -D SuperLUDist_INCLUDE_DIRS:PATH="$TPLDIR/include" \
      -D SuperLUDist_LIBRARY_DIRS:PATH="$TPLDIR/lib" \
      -D SuperLUDist_LIBRARY_NAMES:STRING="superlu_dist_2.5" \
    $EXTRA_ARGS \
    ${TRILINOS_HOME}
    ```
    </details>
1. Build log: 
    <details><summary>Build error</summary>

    ```
    In file included from /ssd/codes/mayrmt_trilinos/src-trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_ParameterListInterpreter.cpp:49:0:
    /ssd/codes/mayrmt_trilinos/src-trilinos/packages/muelu/src/Interface/MueLu_ParameterListInterpreter_def.hpp: In member function ‘void MueLu::ParameterListInterpreter<Scalar, LocalOrdinal, GlobalOrdinal, Node>::UpdateFactoryManager_Repartition(Teuchos::ParameterList&, const Teuchos::ParameterList&, MueLu::ParameterListInterpreter<Scalar, LocalOrdinal, GlobalOrdinal, Node>::FactoryManager&, int, std::vector<std::pair<std::__cxx11::basic_string<char>, const MueLu::FactoryBase*> >&, Teuchos::RCP<MueLu::Factory>&) const’:
    /ssd/codes/mayrmt_trilinos/src-trilinos/packages/muelu/src/Interface/MueLu_ParameterListInterpreter_def.hpp:1369:37: error: ‘switched’ was not declared in this scope
       if (partName == "zoltan2" && !switched) {
    ```
    </details>

--- a/packages/muelu/src/Interface/MueLu_ParameterListInterpreter_def.hpp
+++ b/packages/muelu/src/Interface/MueLu_ParameterListInterpreter_def.hpp
@@ -904,7 +904,7 @@ namespace MueLu {
         "SPARSE BLOCK RELAXATION", "SPARSE_BLOCK_RELAXATION", "SPARSEBLOCKRELAXATION",
         "LINESMOOTHING_BANDEDRELAXATION", "LINESMOOTHING_BANDED_RELAXATION", "LINESMOOTHING_BANDED RELAXATION",
         "LINESMOOTHING_TRIDIRELAXATION", "LINESMOOTHING_TRIDI_RELAXATION", "LINESMOOTHING_TRIDI RELAXATION",
-        "LINESMOOTHING_TRIDIAGONALRELAXATION", "LINESMOOTHING_TRIDIAGONAL_RELAXATION", "LINESMOOTHING_TRIDIAGONAL RELAXATION",             
+        "LINESMOOTHING_TRIDIAGONALRELAXATION", "LINESMOOTHING_TRIDIAGONAL_RELAXATION", "LINESMOOTHING_TRIDIAGONAL RELAXATION",
         "TOPOLOGICAL", "FAST_ILU", "FAST_IC", "FAST_ILDL"}).count(coarseType)) {
         coarseSmoother = rcp(new TrilinosSmoother(coarseType, coarseParams, overlap));
       } else {
@@ -1280,7 +1280,7 @@ namespace MueLu {
     // === Restriction: Nullspace Scaling ===
     if (paramList.isParameter("restriction: scale nullspace") && paramList.get<bool>("restriction: scale nullspace")) {
       RCP<TentativePFactory> tentPFactory = rcp(new TentativePFactory());
-      Teuchos::ParameterList tentPlist;  
+      Teuchos::ParameterList tentPlist;
       tentPlist.set("Nullspace name","Scaled Nullspace");
       tentPFactory->SetParameterList(tentPlist);
       tentPFactory->SetFactory("Aggregates",manager.GetFactory("Aggregates"));
@@ -1290,7 +1290,7 @@ namespace MueLu {
       R->SetFactory("P",tentPFactory);
     }
 
- 
+
   }
 
   // =====================================================================================================
@@ -1361,7 +1361,7 @@ namespace MueLu {
         switched = true;
       }
 #else
-# ifndef HAVE_MUELU_ZOLTAN
+# ifndef HAVE_MUELU_ZOLTAN2
       bool switched = false;
 # endif
 #endif
@@ -1419,7 +1419,7 @@ namespace MueLu {
         throw Exceptions::RuntimeError("Zoltan2 interface is not available");
 #endif
       }
-     
+
       partitioner->SetFactory("A",                    manager.GetFactory("A"));
       partitioner->SetFactory("number of partitions", manager.GetFactory("number of partitions"));
       if (useCoordinates_)

